compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -109,7 +104,8 @@ void addGPUMatmulSimtPassPipeline(OpPassManager &pm) {
   pm.addNestedPass<func::FuncOp>(createGPUPipeliningPass());
 }
 
-void addGPUMatmulTensorCorePassPipeline(OpPassManager &pm) {
+void addGPUMatmulTensorCorePassPipeline(OpPassManager &pm,
```

Comment:
Is it possible to just change the pass that is using the `pipelineDepth` to query the information from the `hal.executable.entry_point`. You still have access to that in the pass. The point of having the attribute is to avoid passing things around in C++.

Reply:
@nirvedhmeshram and I looked at this solution but this turns out to be painful as `updateTranslationInfoAttr` overrides the TranslationInfoAttr. We could update this function and read the value from the entry point but at this point I'm not sure it is worth adding code for that. It looks like the attributes are not scaling as adding one attributes requires writing a significant amount of code.
I think we won't want to add more attribute until we get a better solution and I would rather keep the current logic of the code unchanged. The current solution works and requires less code change so I think we should keep it. @nirvedhmeshram please add a comment explaining the problem so that it is clear why it is done that way and we know this needs to be addressed.

Reply:
Hi Mahesh so I spent some time doing this but we hit a problem and debuged it with @ThomasRaoux and  we found that the attributes were broken here, https://github.com/google/iree/blob/fe42debb5448e416f63490d71d0530ba1930a35e/compiler/src/iree/compiler/Codegen/Common/InsertDistributionInfoPass.cpp#L121
Basically got
``` InsertDistibutionPassBefore
#iree_codegen.translation_info<LLVMGPUMatmulTensorCore pipeline_depth = 2>
 InsertDistibutionPassAfter
#iree_codegen.translation_info<LLVMGPUMatmulTensorCore workload_per_wg = [32, 32]>```
We could fix this but since the solution would be ad hoc we decided rather just pass it through C++ for now where we save the value earlier on

Reply:
DejaVu again. I have acknowledged the problem of attributes and then the only other option is to use the transform dialect. If the cuda backend wants to stay in its own island, then that is going to be a problem. 

 I think you might be over thinking the issue. I don't see why you need to update the translation info attribute. It is set once and then just read. That should be easy and is done without any issue on the CPU side.

Reply:
Ok, the issue is a choice made on the CUDA backend. The workload_per_wg attribute is actually redundant with the tile sizes. So this hack that is in the `InsertDistributionInfoPass.cpp` is just to keep its usage in the CUDA backend (I dont remember the specifics right now). I see why its causing a problem for you here. It should be removed. I am happy to drop this hack in the `InsertDistributionInfoPass.cpp` if the use of `workload_per_wg` is dropped from the CUDA backend.

Reply:
I already forgot the `workload_per_wg` story. You are saying the CPU backend doesn't use it. I can spend some time removing it if other backends have already moved on. Is the alternative to just calculate it on the fly where needed with something like this:
`  SmallVector<int64_t> workloadPerWorkroup = getWorkloadPerWorkgroup(tileSizes);`
?

Reply:
If I remember correctly the only load bearing place they are used is in the pattern to remove the loops based on tile size and loop bounds. If there is a way to drop the usage there, then we can totally drop `workload_per_wg`. It is completely redundant with the `tile_size`.

Reply:
I see it used there and in SetNumWorkgroups. Sounds like we could just recalculate there? I can do that next week